### DISCUSSION TITLE
Functions DB?

### DISCUSSION THREAD

**Discussion by Jebiel** [MeetKai/functionary#147] [General]
If I understand correctly, all the tools that I define, will be injected, as part of the context for the conversation.

What is the limit on the number of functions, before it gets critical?

Also, has anyone tried the approach of defining functions in a vector db? And inject the function schemas as needed, to reduce the number of tokens?

Best regards

**Comment by musabgultekin** [comment]
There is no limit on tool definitions aside from context limit. You can
define as much as you want as long as it fits into the max context length.

When you say defining functions in a vector db, what do you mean exactly?

**Reply by Jebiel** [reply]
That's exactly what I meant 😊

So if I understand correctly, let's say I base my calculations on the get_current_weather schema from the example, and the fact that there's an 8k context.

I used the OpenAI tokenizer, and I get the weather function schema to around 96 tokens.

That's 8000 / 96 ≈ 83 functions.

And of course less than this, since I also need room for my own prompt.

That's why I was thinking, imagine I have 200 functions, is there any way I could save the function schemas in a vector db, and only include them as needed, in my context?

Maybe that has already been done before?
